### ISSUE TITLE
How to specify barcode from 2+ reads  with `DemuxFastqs`

### ISSUE THREAD

**Issue by gshiba** [fulcrumgenomics/fgbio#310]
I'd like to use `DemuxFastqs` for a library where the barcode is split over 2 or more reads, but I can't seem to get it to work.

**Should the `Sample_Barcode` column in metadata.csv include dashes? `ACGTACGT` vs `ACGT-ACGT`**

```
$ fgbio --version  # conda installation
Version: 0.2.1-SNAPSHOT

$ cat metadata.csv
Sample_ID,Sample_Name,Library_ID,Description,Sample_Barcode
one,nice sample one,lib_one,very nice sample,GGTCGACA

$ fgbio DemuxFastqs --inputs r1.fastq.gz i1.fastq.gz i2.fastq.gz r2.fastq.gz --read-structures +T 4B+S 4B+S +T --output demuxed/ --metadata metadata.csv
...
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: The number of sample barcodes bases did not match; read structures: 8 sample (unmatched): 9
        at scala.Predef$.require(Predef.scala:277)
        at com.fulcrumgenomics.fastq.DemuxFastqs.$anonfun$execute$6(DemuxFastqs.scala:352)
        at com.fulcrumgenomics.fastq.DemuxFastqs.$anonfun$execute$6$adapted(DemuxFastqs.scala:349)
        at scala.collection.immutable.List.foreach(List.scala:389)
        at com.fulcrumgenomics.fastq.DemuxFastqs.execute(DemuxFastqs.scala:349)
        at com.fulcrumgenomics.cmdline.FgBioMain.makeItSo(FgBioMain.scala:93)
        at com.fulcrumgenomics.cmdline.FgBioMain.makeItSoAndExit(FgBioMain.scala:74)
        at com.fulcrumgenomics.cmdline.FgBioMain$.main(FgBioMain.scala:45)
        at com.fulcrumgenomics.cmdline.FgBioMain.main(FgBioMain.scala)

$ vi metadata.csv  # insert dash in barcode sequence
$ cat metadata.csv
Sample_ID,Sample_Name,Library_ID,Description,Sample_Barcode
one,nice sample one,lib_one,very nice sample,GGTC-GACA

$ fgbio DemuxFastqs --inputs r1.fastq.gz i1.fastq.gz i2.fastq.gz r2.fastq.gz --read-structures +T 4B+S 4B+S +T --output demuxed/ --metadata metadata.csv
...
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: The number of sample barcodes bases did not match; read structures: 8 sample (one): 9
...
```

Relevant lines from the source code:

The (automatic) unmatched sample gets a `-` (dash):
https://github.com/fulcrumgenomics/fgbio/blob/d0c1454eff9a8bd6a32debc3f873cdb997a44251/src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala#L115

, but it's `require`d against the `--read-structures` without a dash:
https://github.com/fulcrumgenomics/fgbio/blob/d0c1454eff9a8bd6a32debc3f873cdb997a44251/src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala#L348

Thanks

**Comment by tfenne** [comment]
Hi @gshiba I think this is a bug in the tool.  Your first attempt (just concatenating the bases) is what the tool documentation describes, and should work just fine.

I think the problem is that internally the tool generates an additional sample entry for the unmatched sample, and it's not following it's own instructions.  It appears to generate it by constructing strings of Ns matching the lengths of the barcode reads, and then concatenates them with `-`s between, when it should presumably not be putting hypens in.

@nh13 Does that sound right to you?  If so, it's an easy fix.

**Comment by nh13** [comment]
@gshiba fixed in #310.  Thank-you for the very clear bug report and debugging!
